src/Facets/GenericSwapFacet.sol

```diff
@@ -3,42 +3,232 @@ pragma solidity 0.8.17;
+        string calldata _integrator,
+        string calldata _referrer,
+        address payable _receiver,
+        uint256 _minAmountOut,
+        LibSwap.SwapData calldata _swapData
+    ) external payable {
+        _depositAndSwapERC20(_swapData);
+
+        // get contract's balance (which will be sent in full to user)
+        uint256 amountReceived = ERC20(_swapData.receivingAssetId).balanceOf(
+            address(this)
+        );
+
+        // ensure that minAmountOut was received
+        if (amountReceived < _minAmountOut)
+            revert CumulativeSlippageTooHigh(_minAmountOut, amountReceived);
+
+        // transfer funds to receiver
+        ERC20(_swapData.receivingAssetId).safeTransfer(
+            _receiver,
+            amountReceived
+        );
+
+        // emit events (both required for tracking)
+        emit LibSwap.AssetSwapped(
+            _transactionId,
+            _swapData.callTo,
+            _swapData.sendingAssetId,
+            _swapData.receivingAssetId,
+            _swapData.fromAmount,
+            amountReceived,
+            block.timestamp
+        );
+
+        emit LiFiGenericSwapCompleted(
+            _transactionId,
+            _integrator,
+            _referrer,
+            _receiver,
+            _swapData.sendingAssetId,
+            _swapData.receivingAssetId,
+            _swapData.fromAmount,
+            amountReceived
+        );
+    }
+
+    /// @notice Performs a single swap from an ERC20 token to the network's native token
+    /// @param _transactionId the transaction id associated with the operation
+    /// @param _integrator the name of the integrator
+    /// @param _referrer the address of the referrer
+    /// @param _receiver the address to receive the swapped tokens into (also excess tokens)
+    /// @param _minAmountOut the minimum amount of the final asset to receive
+    /// @param _swapData an object containing swap related data to perform swaps before bridging
+    function swapTokensSingleERC20ToNative(
+        bytes32 _transactionId,
+        string calldata _integrator,
+        string calldata _referrer,
+        address payable _receiver,
+        uint256 _minAmountOut,
+        LibSwap.SwapData calldata _swapData
+    ) external payable {
+        _depositAndSwapERC20(_swapData);
+
+        // get contract's balance (which will be sent in full to user)
+        uint256 amountReceived = address(this).balance;
+
+        // ensure that minAmountOut was received
+        if (amountReceived < _minAmountOut)
+            revert CumulativeSlippageTooHigh(_minAmountOut, amountReceived);
+
+        // transfer funds to receiver
+        // solhint-disable-next-line avoid-low-level-calls
+        (bool success, ) = _receiver.call{ value: amountReceived }("");
+        if (!success) revert NativeAssetTransferFailed();
+
+        // emit events (both required for tracking)
+        emit LibSwap.AssetSwapped(
+            _transactionId,
```

Comment:
nit: I would move this to the internal `_depositAndSwapERC20` method to be more consistent with the rest of the codebase.

Reply:
Can't do this as it needs the amountOut which varies per function.